internal/commands/internal/test/test.go

```diff
@@ -96,27 +97,43 @@ func test(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("%w: %v", ErrTest, err)
 		}
 
-		diff := deep.Equal(expectedConfig, generatedConfig)
-		currentTestSucceeded := len(diff) == 0
+		differentConfig := logDifferenceIfAny(logger, expectedConfig, generatedConfig)
 
-		if !currentTestSucceeded {
-			for _, line := range diff {
-				logger.Warnf("%s", line)
-			}
+		logsBytes, err := ioutil.ReadFile(filepath.Join(testDir, ".cirrus.expected.log"))
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("%w: %v", ErrTest, err)
+		}
+		differentLogs := logDifferenceIfAny(
+			logger,
+			strings.Split(string(logsBytes), "\n"),
+			strings.Split(string(result.OutputLogs), "\n"),
+		)
+
+		if differentConfig || differentLogs {
 			someTestsFailed = true
 		}
 
-		logger.Finish(currentTestSucceeded)
+		logger.Finish(!differentConfig && !differentLogs)
 	}
 
-	logger.Finish(someTestsFailed)
+	logger.Finish(!someTestsFailed)
 	if someTestsFailed {
 		return fmt.Errorf("%w: some tests failed", ErrTest)
 	}
 
 	return nil
 }
 
+func logDifferenceIfAny(logger *echelon.Logger, a, b interface{}) bool {
```

Comment:
There needs to be a way to differentiate between differences somehow, since there might be a test that needs to check both the config and logs for differences.

The output for a such test would look something like this:

```
❌ internal/commands/internal/test/testdata/logs 0.0s
   Kind: 1 != 0
   Content: [0xc000513c20] != <nil slice>
   Line: 1 != 0
   Column: 1 != 0
   slice[1]: 123 != 
   slice[2]:  != <no value>
```

Reply:
What is your suggestion? This weird output seems to be from the initial comparison of raw `Node`s.

Reply:
>What is your suggestion?

```patch
diff --git a/internal/commands/internal/test/test.go b/internal/commands/internal/test/test.go
index 44ce841..b441707 100644
--- a/internal/commands/internal/test/test.go
+++ b/internal/commands/internal/test/test.go
@@ -97,13 +97,14 @@ func test(cmd *cobra.Command, args []string) error {
                        return fmt.Errorf("%w: %v", ErrTest, err)
                }
 
-               differentConfig := logDifferenceIfAny(logger, expectedConfig, generatedConfig)
+               differentConfig := logDifferenceIfAny("YAML configuration", logger, expectedConfig, generatedConfig)
 
                logsBytes, err := ioutil.ReadFile(filepath.Join(testDir, ".cirrus.expected.log"))
                if err != nil && !errors.Is(err, os.ErrNotExist) {
                        return fmt.Errorf("%w: %v", ErrTest, err)
                }
                differentLogs := logDifferenceIfAny(
+                       "logs",
                        logger,
                        strings.Split(string(logsBytes), "\n"),
                        strings.Split(string(result.OutputLogs), "\n"),
@@ -124,9 +125,11 @@ func test(cmd *cobra.Command, args []string) error {
        return nil
 }
 
-func logDifferenceIfAny(logger *echelon.Logger, a, b interface{}) bool {
+func logDifferenceIfAny(where string, logger *echelon.Logger, a, b interface{}) bool {
        diff := deep.Equal(a, b)
 
+       logger.Warnf("detected difference in %s:", where)
+
        for _, line := range diff {
                logger.Warnf("%s", line)
        }
```

Reply:
See 5a6df8777828aee7b801c276eea6e4f6f68337c1